packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -449,6 +346,50 @@ const EDiscovery = WebexPlugin.extend({
     return '';
   },
 
+  _isUrl(string) {
+    return /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&\/\/=]*)/g.test(string);
+  },
+
+  /**
+   * Create the appropriate request options based on whether the reportId is a url or a uuid.
+   * @param {String} reportId - May be either a url for the report or a uuid.
+   * e.g. 'http://ediscovery-intb.wbx2.com/ediscovery/api/v1/reports/3b10e625-2bd5-4efa-b866-58d6c93c505c' or '3b10e625-2bd5-4efa-b866-58d6c93c505c'.
+   * @param {String} resource - The resource on the remote server to request
+   * @returns {[Options, uuid]} - The options to pass to `request` and the report uuid.
+   */
+  _createRequestOptions(reportId, resource) {
+    let requestOptions;
+    let reportUUID;
+    const isUrl = this._isUrl(reportId);
+
+    if (isUrl) {
+      reportUUID = this._getReportIdFromUrl(reportId);
+
+      if (!reportUUID) {
+        throw Error('Report url does not contain a report id');
+      }
+
+      // Ensure the url is formated to
+      // https://ediscovery.intb1.ciscospark.com/ediscovery/api/v1/reports/16bf0d01-b1f7-483b-89a2-915144158fb9
+      // index.js for example passes the url in as
+      // https://ediscovery.intb1.ciscospark.com/ediscovery/api/v1/reports/16bf0d01-b1f7-483b-89a2-915144158fb9/contents
+      const reportUrl = reportId.substring(0, reportId.lastIndexOf(reportUUID) + reportUUID.length);
+
+      requestOptions = {
+        url: reportUrl + resource
```

Comment:
I'd move the substring function in here, but the comment is so large that keeping it separate is probably best